packages/devtools_app/lib/src/memory/flutter/memory_chart.dart

```diff
@@ -0,0 +1,342 @@
+      final y = externalMemoryData[index + 1];
+
+      _externalHeap.add(Entry(x: x.toDouble(), y: y.toDouble(), icon: _img));
+      index += 2;
+    }
+
+    index = 0;
+    while (index < usedHeapData.length) {
+      final x = usedHeapData[index];
+      final y = usedHeapData[index + 1] + externalMemoryData[index + 1];
+
+      _used.add(Entry(x: x.toDouble(), y: y.toDouble(), icon: _img));
+      index += 2;
+    }
+
+    index = 0;
+    while (index < heapCapacityData.length) {
+      final x = heapCapacityData[index];
+      final y = heapCapacityData[index + 1];
+
+      _capacity.add(Entry(x: x.toDouble(), y: y.toDouble(), icon: _img));
+      index += 2;
+    }
+  }
+
+  void _initLineData([bool simulateFeed = false]) async {
+    if (!simulateFeed) {
+      await loadAllData();
+    } else if (_timer == null) {
+      await startFeed();
+    } else {
+      return;
+    }
+
+    // Create heap used dataset.
+    usedHeapSet = LineDataSet(_used, 'Used');
+    usedHeapSet
+      ..setAxisDependency(AxisDependency.LEFT)
+      ..setColor1(ColorUtils.getHoloBlue())
+      ..setValueTextColor(ColorUtils.getHoloBlue())
+      ..setLineWidth(.7)
+      ..setDrawCircles(false)
+      ..setDrawValues(false)
+      ..setFillAlpha(65)
+      ..setFillColor(ColorUtils.getHoloBlue())
+      ..setDrawCircleHole(false)
+      // Fill in area under set.
+      ..setDrawFilled(true)
+      ..setFillColor(ColorUtils.getHoloBlue())
+      ..setFillAlpha(80);
+
+    // Create heap capacity dataset.
+    capacityHeapSet = LineDataSet(_capacity, 'Capacity')
+      ..setAxisDependency(AxisDependency.LEFT)
+      ..setColor1(ColorUtils.GRAY)
+      ..setValueTextColor(ColorUtils.GRAY)
+      ..setLineWidth(.5)
+      ..enableDashedLine(5, 5, 0)
+      ..setDrawCircles(false)
+      ..setDrawValues(false)
+      ..setFillAlpha(65)
+      ..setFillColor(ColorUtils.GRAY)
+      ..setDrawCircleHole(false);
+
+    // Create external memory dataset.
+    const externalColorLine =
+        Color.fromARGB(0xff, 0x42, 0xa5, 0xf5); // Color.blue[400]
+    const externalColor =
+        Color.fromARGB(0xff, 0x90, 0xca, 0xf9); // Color.blue[200]
+    externalMemorySet = LineDataSet(_externalHeap, 'External');
+    externalMemorySet
+      ..setAxisDependency(AxisDependency.LEFT)
+      ..setColor1(externalColorLine)
+      ..setLineWidth(.7)
+      ..setDrawCircles(false)
+      ..setDrawValues(false)
+      ..setHighLightColor(const Color.fromARGB(255, 244, 117, 117))
+      ..setDrawCircleHole(false)
+      // Fill in area under set.
+      ..setDrawFilled(true)
+      ..setFillColor(externalColor)
+      ..setFillAlpha(190);
+
+    // Create a data object with all the data sets.
+    _chartController.data = LineData.fromList(
+        []..add(usedHeapSet)..add(externalMemorySet)..add(capacityHeapSet));
+
+    _chartController.data
+      ..setValueTextColor(ColorUtils.getHoloBlue())
+      ..setValueTextSize(9);
+
+    setState(() {});
```

Comment:
Because this is called from initState, the setState here shouldn't be necessary.

Reply:
Good point - its gone.